Sources/NIOWebSocket/WebSocketFrameEncoder.swift

```diff
@@ -16,7 +16,14 @@ import NIO
 
 private let maxOneByteSize = 125
 private let maxTwoByteSize = Int(UInt16.max)
-private let maxNIOFrameSize = Int(UInt32.max)
+#if arch(arm) // 32-bit, Raspi/AppleWatch/etc
+    // Note(hh): This is not a _proper_ fix, but necessary because
+    //           other places extend on that. Should be fine in
+    //           practice on 32-bit platforms.
+    private let maxNIOFrameSize = Int(Int32.max / 4)
```

Comment:
Any reason not to just set this to `Int(Int32.max )`? So far as I know we don't ever do math on this, so setting it to `Int32.max` gets the behaviour as close as reasonably possible to the 64-bit behaviour.

Reply:
Same as above. Probably breaks when it travels to the ByteBuffer, which will use UInt32 and then fail.

Reply:
we should fully understand this. I think `Int32.max` should just work